parser/mpFuncStr.cpp

```diff
@@ -138,6 +138,123 @@ MUP_NAMESPACE_START
     return new FunStrRight(*this);
   }
 
+  //------------------------------------------------------------------------------
+  //
+  // default_value() auxiliary overloading functions
+  //
+  //------------------------------------------------------------------------------
+  int_type default_value(int_type value, int_type standard) {
+    return value == NULL ? standard : value;
+  }
+
+  float_type default_value(float_type value, float_type standard) {
+    return value == NULL ? standard : value;
+  }
+
+  string_type default_value(string_type value, string_type standard) {
+    return value.empty() ? standard : value;
+  }
+
+  // If the value is an integer, it is NULL, therefore return standard
+  string_type default_value(int_type value, string_type standard) {
+    return standard;
+  }
+
+  bool_type default_value(bool_type value, bool_type standard) {
+    return value;
+  }
+
+  // If the value is an integer, it is NULL, therefore return standard
+  bool_type default_value(int_type value, bool_type standard) {
+    return standard;
+  }
+
+  //------------------------------------------------------------------------------
+  //
+  // Default Value function
+  //
+  //------------------------------------------------------------------------------
+
+  FunStrDefaultValue::FunStrDefaultValue()
+    :ICallback(cmFUNC, _T("default_value"), 2)
+  {}
+
+  //------------------------------------------------------------------------------
+  void FunStrDefaultValue::Eval(ptr_val_type &ret, const ptr_val_type *a_pArg, int)
+  {
+
+    int_type integer_value;
+    int_type integer_standard;
+    float_type float_value;
+    float_type float_standard;
+    string_type string_value;
+    string_type string_standard;
+    bool_type bool_value;
+    bool_type bool_standard;
+
+    if (a_pArg[0]->GetType() != a_pArg[1]->GetType()) {
+      if (a_pArg[0]->GetInteger() != NULL) {
+        throw ParserError(ErrorContext(ecINVALID_TYPES_MATCH, GetExprPos(), GetIdent()));
+      }
+    }

```

Comment:
Avoid different types for the parameters
```rb
default_value(5, "default")
```

But allow if the first parameter is `NULL`
```rb
default_value(NULL, "default")
```